### DISCUSSION TITLE
Possible options for parsing the field value when sending it to the mail

### DISCUSSION THREAD

**Discussion by Firiate** [jertel/elastalert2#1283] [Q&A] [answered]
What options are possible for parsing the field value. Example of a field that is sent to the mail:

event.original: <190>date=2023-09-28 time=09:41:37 devname=main-fw01-node1 device_id=GF8894098444 log_id=005678976 type=event subtype=dm pri=information desc="Policy package install failed" user="john.smith" userfrom="" msg="Installation of the policy package main-fw01_Main on main-fw01[Main] failed." adom="root" session_id=145675 instpkg="main-fw01_Main"

Is it possible to use the means of an elastalert (for example - regular expressions) to send only user="john.smith" and instpkg="main-fw01_Main"

Thank you.

**Comment by jertel** [comment, marked as answer]
You can either adjust your ingest pipelines to do the parsing there, or you can add a custom [enhancement](https://elastalert2.readthedocs.io/en/latest/recipes/adding_enhancements.html#enhancements) to your ElastAlert 2 installation, using a few lines of Python code. There are a number of discussions here you can look through to see what problems others have solved using enhancements.
